File: src/main/java/cz/grunwjir/csasdemo/csas/client/BranchData.java
package cz.grunwjir.csasdemo.csas.client;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

/**
 * Data object that represents Csas branch.
 *
 * @author Jiri Grunwald
 */
@Data
public class BranchData {
    private Integer id;
    private String name;
    private String address;
    private String city;
    private String postCode;
    private String managerName;
    private List<String> phones;
    private String email;
    @JsonProperty("equipment")
    private List<Equipment> equipments;

    @Data
    public static class Equipment {
        private Integer id;
        private String name;
        private Integer count;
    }
}

File: src/main/kotlin/cz/grunwjir/csasdemo/csas/client/BranchData.kt
package cz.grunwjir.csasdemo.csas.client

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Data object that represents Csas branch.
 *
 * @author Jiri Grunwald
 */
data class BranchData(val id: Int,
                      val name: String,
                      val address: String,
                      val city: String,
                      val postCode: String,
                      val managerName: String?,
                      val phones: List<String>?,
                      val email: String?,
                      @JsonProperty("equipment") val equipments: List<Equipment>?) {

    data class Equipment(val id: Int,
                         val name: String,
                         val count: Int)
}